app/models/tag.rb

```diff
@@ -61,17 +55,29 @@ def all_available_tags
 
   private
 
-  def create_cognates(str_list)
-    str_list.filter_map do |name|
-      next if name.blank?
+  def create_cognates(names)
+    names.each { |name| Tag.find_or_create_by(name: name) }
```

Comment:
We can detect here those tags that already exist and (I guess) were associated previously, leaving only new names that should be processed.
Something like this:
```ruby
names
  .map { |name| [name, Tag.find_or_initialize_by(name: name)] }
  .each_with_object([]) do |(name, tag), new_names|
    new_names << name unless tag.persists?
    tag.save
    new_names
  end
```
This should create missing tags and return names for these tags. So we can pass these names into `associate_cognates` method

Sorry if I misunderstood something